src/lock_file/pypi.rs

```diff
@@ -82,13 +83,23 @@ pub async fn resolve_dependencies<'db>(
     };
 
     // Resolve the PyPi dependencies
+    let mapping = conda_pypi_name_mapping().await?;
     let mut result = resolve(
         package_db,
         &requirements,
         &marker_environment,
         Some(&compatible_tags),
         conda_python_packages
             .into_iter()
+            // skip using the locked package in the case where conda and pypi package has the same name
+            .filter(|p| {
+                let has_different_name = requirements.iter().any(|r| r.name != p.name.as_str());
+                let found_in_mapping = mapping.get(p.name.as_str()).is_some();
+                if !has_different_name {
+                    tracing::warn!("{} exists in both Conda and PyPI dependencies", p.name);
+                }
+                has_different_name || found_in_mapping
```

Comment:
Sorry i had this review sitting for some time, without publishing it.....

Reply:
Oh, so just revert the condition or do something else? Yeah, definitely need tests :/

Reply:
I just tried again and the behavior is the same for me. Can you maybe clean up `.pixi` before trying? There might be some stuff going on with the satisfiability checks.

Also I just checked the tests and I think it would make sense to add an integration test for this case instead of creating an unit test. Will work on that.

Reply:
One other question, should we also update the lock file when we run `pixi add`? i.e. when we run `pixi add --pypi pandoc` do we expect to see `pandoc` in the lock file as PyPI dependency? Currently that's not the case.

I'm thinking of adding a test like this:

```rust
let pixi = PixiControl::new().unwrap();

pixi.init().await.unwrap();

pixi.add("python")
    .set_type(DependencyType::CondaDependency(SpecType::Run))
    .with_install(false)
    .await
    .unwrap();

pixi.add("pandoc")
    .set_type(DependencyType::CondaDependency(SpecType::Build))
    .await
    .unwrap();
pixi.add("pandoc")
    .set_type(DependencyType::PypiDependency)
    .with_install(true)
    .await
    .unwrap();
```

Then simply asserting the contents of the lock file for PyPI dependency. I'm not sure if this is a good idea or there is a better way to test it. LMK @ruben-arts!

Reply:
Yeah the with install should add it to the lockfile. If you do the same for numpy you want to see numpy only once in the lockfile and pandoc twice.